### PR TITLE
Add DNX_REQUIRE_SERVICING to prevent DNX from starting without servicing

### PR DESCRIPTION
**Pull request by victorhurdugaci**
Fixes: https://github.com/aspnet/dnx/issues/1122

@moozzyk @davidfowl 
fyi @blowdart

### CODE REVIEW

**Comment by victorhurdugaci** [comment]
Helios change coming soon

File: src/dnx/servicing.cpp [hunk 1]
```diff
@@ -70,7 +70,26 @@ namespace dnx
...
+            int require_servicing_buffer_length = GetEnvironmentVariable(L"DNX_REQUIRE_SERVICING", require_servicing_buffer, 2);
+
+            if (require_servicing_buffer_length > 1 ||
+                (require_servicing_buffer_length == 1 && (require_servicing_buffer[0] != L'1' && require_servicing_buffer[0] != L'0')))
+            {
+                throw std::runtime_error("The value of the DNX_REQUIRE_SERVICING environment variable is invalid. Accepted values: '0' or '1'. The application will now exit.");
+            }
+
+            bool require_servicing = require_servicing_buffer_length != 0 && require_servicing_buffer[0] == L'1';
+            bool servicing_root_exists = utils::directory_exists(servicing_root);
```
Comment (moozzyk) [inline src/dnx/servicing.cpp]: nit use 'auto'

File: src/dnx/servicing.cpp [hunk 2]
```diff
@@ -70,7 +70,26 @@ namespace dnx
                 ? utils::path_combine(servicing_root_parent, std::wstring(L"Microsoft DNX\\Servicing"))
                 : servicing_root_parent;
 
-            if (is_default_servicing_location && !utils::directory_exists(servicing_root))
+            wchar_t require_servicing_buffer[2] = { 0 , 0 };
```
Comment (moozzyk) [inline src/dnx/servicing.cpp]: please move lines 73-82 to a separate function that returns bool (or throws).

**Comment by moozzyk** [comment]
:ship::it: (after the changes above)

File: src/dnx/servicing.cpp [hunk 3]
```diff
@@ -70,7 +70,26 @@ namespace dnx
...
+            int require_servicing_buffer_length = GetEnvironmentVariable(L"DNX_REQUIRE_SERVICING", require_servicing_buffer, 2);
+
+            if (require_servicing_buffer_length > 1 ||
+                (require_servicing_buffer_length == 1 && (require_servicing_buffer[0] != L'1' && require_servicing_buffer[0] != L'0')))
+            {
+                throw std::runtime_error("The value of the DNX_REQUIRE_SERVICING environment variable is invalid. Accepted values: '0' or '1'. The application will now exit.");
```
Comment (troydai) [inline src/dnx/servicing.cpp]: This doesn't seem to be a common behaviour. Usually if the environment variable value doesn't fit the needs we just don't set the switch. Throw exception seems extreme.

**Comment by troydai** [comment]
:shipit: my comments shouldn't block the PR.

**Comment by victorhurdugaci** [comment]
Updated

File: src/dnx/servicing.cpp [hunk 4]
```diff
@@ -64,13 +64,36 @@ namespace dnx
...
         {
             auto servicing_root = is_default_servicing_location
                 ? utils::path_combine(servicing_root_parent, std::wstring(L"Microsoft DNX\\Servicing"))
                 : servicing_root_parent;
 
-            if (is_default_servicing_location && !utils::directory_exists(servicing_root))
+            auto servicing_root_exists = utils::directory_exists(servicing_root);
+            auto require_servicing = get_require_servicing();
+                     
+            if (require_servicing && !servicing_root_exists)
```
Comment (moozzyk) [inline src/dnx/servicing.cpp]: consider nesting these two conditions under single `if (!servicig_root_exists)` instead of comparing it twice.

**Comment by moozzyk** [comment]
It's good to go.

**Pull request merged**